lib/chef_zero/endpoints/cookbook_version_endpoint.rb

```diff
@@ -85,11 +85,13 @@ def get_checksums(cookbook)
 
       private
 
-      def hoover_unused_checksums(deleted_checksums, request, data_type='cookbooks')
-        data_store.list(request.rest_path[0..1] + [data_type]).each do |cookbook_name|
-          data_store.list(request.rest_path[0..1] + [data_type, cookbook_name]).each do |version|
-            cookbook = data_store.get(request.rest_path[0..1] + [data_type, cookbook_name, version], request)
-            deleted_checksums = deleted_checksums - get_checksums(cookbook)
+      def hoover_unused_checksums(deleted_checksums, request)
+        %w(cookbooks cookbook_artifacts).each do |cookbook_type|
+          data_store.list(request.rest_path[0..1] + [cookbook_type]).each do |cookbook_name|
+            data_store.list(request.rest_path[0..1] + [cookbook_type, cookbook_name]).each do |version|
+              cookbook = data_store.get(request.rest_path[0..1] + [cookbook_type, cookbook_name, version], request)
+              deleted_checksums = deleted_checksums - get_checksums(cookbook)
+            end
```

Comment:
So this does "don't delete anything still in use", but doesn't do "delete anything _*not*_ still in use (i.e. orphans)", which isn't a change in behavior, but something to keep in mind.